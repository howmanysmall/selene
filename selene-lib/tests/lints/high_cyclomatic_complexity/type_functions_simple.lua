-- Test that TypeFunction and ExportedTypeFunction don't cause compilation errors
-- This file doesn't actually contain type functions since they may not be parseable
-- The real test is that the code compiles and doesn't crash

-- Regular function should still contribute to complexity (this should warn)
local function regular_function()
    if true then
        if true then
            if true then
                if true then
                    if true then
                        if true then
                            if true then
                                if true then
                                    if true then
                                        if true then
                                            if true then
                                                return "high complexity"
                                            end
                                        end
                                    end
                                end
                            end
                        end
                    end
                end
            end
        end
    end
end
