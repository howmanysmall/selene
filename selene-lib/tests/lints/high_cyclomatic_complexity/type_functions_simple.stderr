error[high_cyclomatic_complexity]: this function has a cyclomatic complexity of 11
   ┌── type_functions_simple.lua:7:1 ───
   │
 7 │ local function regular_function()
   │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │
   = note: consider refactoring this function
